packages/inference/src/lib/makeRequestOptions.ts

```diff
@@ -81,19 +80,44 @@ export async function makeRequestOptions(
 	if (providerConfig.clientSideRoutingOnly && !maybeModel) {
 		throw new Error(`Provider ${provider} requires a model ID to be passed directly.`);
 	}
+
 	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
 	const hfModel = maybeModel ?? (await loadDefaultModel(task!));
-	const model = skipModelIdResolution
-		? hfModel
-		: providerConfig.clientSideRoutingOnly
-		  ? // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-		    removeProviderPrefix(maybeModel!, provider)
-		  : // For closed-models API providers, one needs to pass the model ID directly (e.g. "gpt-3.5-turbo")
-		    await getProviderModelId({ model: hfModel, provider }, args, {
-					task,
-					chatCompletion,
-					fetch: options?.fetch,
-		    });
+	const resolvedModel = providerConfig.clientSideRoutingOnly
+		? // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		  removeProviderPrefix(maybeModel!, provider)
+		: await getProviderModelId({ model: hfModel, provider }, args, {
+				task,
+				chatCompletion,
+				fetch: options?.fetch,
+		  });
+
+	// Use the sync version with the resolved model
+	return makeRequestOptionsFromResolvedModel(resolvedModel, args, options);
+}
+
+/**
+ * Helper that prepares request arguments. - for internal use only
+ * This sync version skips the model ID resolution step
+ */
+export function makeRequestOptionsFromResolvedModel(
+	resolvedModel: string,
+	args: RequestArgs & {
+		data?: Blob | ArrayBuffer;
+		stream?: boolean;
+	},
+	options?: Options & {
+		task?: InferenceTask;
+		chatCompletion?: boolean;
+	}
+): { url: string; info: RequestInit } {
+	const { accessToken, endpointUrl, provider: maybeProvider, ...remainingArgs } = args;
+	delete remainingArgs.model; // Remove model from remainingArgs to avoid duplication
+
```

Comment:
FYI you can use this syntax


```suggestion
	const { accessToken, endpointUrl, provider: maybeProvider, model, ...remainingArgs } = args;

```

Reply:
Now I remember why I did that in the first place. If I deconstruct `model` and don't use it, the linter complains with `'model' is assigned a value but never used`. 

![image](https://github.com/user-attachments/assets/f40f92ad-170c-4b4f-8bda-ca0aaec801a4)